frontend/src/lib/taxonomy.tsx

```diff
@@ -926,9 +927,42 @@ export function isPostHogProp(key: string): boolean {
 export type PropertyKey = string | null | undefined
 export type PropertyType = 'event' | 'element'
 
+// copy from https://github.com/PostHog/posthog/blob/29ac8d6b2ba5de4b65a148136b681b8e52e20429/plugin-server/src/utils/db/utils.ts#L44
+const eventToPersonProperties = new Set([
+    // mobile params
+    '$app_build',
+    '$app_name',
+    '$app_namespace',
+    '$app_version',
+    // web params
+    '$browser',
+    '$browser_version',
+    '$device_type',
+    '$current_url',
+    '$pathname',
+    '$os',
+    '$os_version',
+    '$referring_domain',
+    '$referrer',
+    // campaign params - automatically added by posthog-js here https://github.com/PostHog/posthog-js/blob/master/src/utils/event-utils.ts
+    'utm_source',
+    'utm_medium',
+    'utm_campaign',
+    'utm_content',
+    'utm_name',
+    'utm_term',
+    'gclid',
+    'gad_source',
+    'gbraid',
+    'wbraid',
+    'fbclid',
+    'msclkid',
+])
+
 export function getKeyMapping(
     value: string | PropertyFilterValue | undefined,
-    type: 'event' | 'element'
+    type: 'event' | 'element',
+    filterGroupType?: TaxonomicFilterGroupType
```

Comment:
This is the comment I was about to post:
> `filterGroupType` seems to pretty much be a superset of `type`, and both are called `*type`, so overall there's a lot of potential for confusion. I propose we do this: `type` becomes `EventProperties.Events | EventProperties.Elements | EventProperties.EventProperties | EventProperties.PersonProperties` (the rest is unsupported).

Then I did that but just that reminded me what a complete mess the whole "keyMapping" system is, in that it just has "events" and… "elements", which doesn't really map to how anything within PosHog works (but probably did 4 years ago).

So instead I thought, why not cut the Gordian knot. The diff ballooned, but _now_ we have a clear mapping here that directly translates to the types of core filters we have in PostHog. And of course we still have those smart "Latest" and "Initial" prefixes for person properties.

Because I did A LOT here, I would love if you also took a good look at these changes @neilkakkar